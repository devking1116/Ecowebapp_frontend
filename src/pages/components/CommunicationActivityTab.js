/* @flow */
import React from 'react';
import { Feed, Comment } from 'semantic-ui-react';
import reactCSS from 'reactcss';

import { Activity } from '../../core';

type Props = {
  activity?: Array<{
    author: string,
    date: string,
    label: string,
  }>,
};

/**
 * The component that displays the activity tab on the Communication Page,
 */
export default function CommunicationActivityTab({ activity, ...otherProps }: Props): React.Element<*> {

  const styles = reactCSS({
    'default': {
      activityWrapper: {
        paddingTop: 20,
      },
    },
  });

  return (
    <Comment.Group threaded  {...otherProps} style={styles.activityWrapper}>
    {activity? activity.map((activityData, index) =>
      <Activity {...activityData} key={index} isLast={(index + 1) ===(activity ? activity.length: 0)? 'true' : 'false'}/>
    ): null}
    </Comment.Group>
  );
}
