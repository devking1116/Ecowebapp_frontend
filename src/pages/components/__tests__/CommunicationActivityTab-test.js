import React from 'react';
import renderer from 'react-test-renderer';

import CommunicationActivityTab from '../CommunicationActivityTab';

const testActivity = [
];

test('it displays the given activity.', () => {
  const component = renderer.create(
    <CommunicationActivityTab activity={testActivity}/>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
