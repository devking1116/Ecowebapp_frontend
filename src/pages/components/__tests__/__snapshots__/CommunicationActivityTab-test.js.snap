exports[`test it displays the given activity. 1`] = `
<div
  className="ui threaded comments"
  style={
    Object {
      "paddingTop": 20,
    }
  } />
`;
